### PR TITLE
2.5 api gateway

### LINKED ISSUE #17: 2.5 API Gateway

### PR DESCRIPTION
**Pull request by magley**
Closes #17 

How to use:
1. Create lambdas with `create_functions.py` in `src/lambdas`
2. Deploy API gateway with `api_gateway_deploy.py` in `scr/lambdas`
3. Update `API_GATEWAY_ID` in `service/session.py` from the id that gets printed out in 2.
4. Run program.

### CODE REVIEW

File: src/lambdas/api_gateway_deploy.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import boto3
+from common_deploy import *
+
+
+def create_rest_api(name: str) -> str:
+    response = apigateway_cli.create_rest_api(
+        name='kalu-drive API',
```
Comment (filipkontic) [inline src/lambdas/api_gateway_deploy.py:7]: `kalu` -> `klau`

File: src/lambdas/api_gateway_deploy.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    )
+
+    apigateway_cli.put_integration(
+        restApiId=rest_api_id,
+        resourceId=resource_id,
+        httpMethod=method,
+        type='AWS_PROXY',
+        integrationHttpMethod='POST',
+        passthroughBehavior='WHEN_NO_MATCH',
+        uri=f'arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/{func_uri}/invocations',
```
Comment (filipkontic) [inline src/lambdas/api_gateway_deploy.py:47]: Replace the `us-east-1` part with the constant `REGION` from `common_deploy.py`. Same for other places in the PR.

**Review by filipkontic — CHANGES_REQUESTED** [review]

File: src/lambdas/api_gateway_deploy.py (hunk 2 above)
Comment (magley) [reply src/lambdas/api_gateway_deploy.py]: [beb0104](https://github.com/magley/klau-drive/pull/20/commits/beb0104866e6900231210de6feb5f82edeb0e660)

File: src/lambdas/api_gateway_deploy.py (hunk 1 above)
Comment (magley) [reply src/lambdas/api_gateway_deploy.py]: [beb0104](https://github.com/magley/klau-drive/pull/20/commits/beb0104866e6900231210de6feb5f82edeb0e660)

**Review by filipkontic — APPROVED** [review]

**Pull request merged**